src/service/okapi.service.ts

```diff
@@ -30,58 +30,48 @@ class OkapiService extends RestService {
         if (!!response && response?.statusCode >= 200 && response?.statusCode <= 299) {
           let accessToken;
           let refreshToken;
-          const headers = !!response?.headers ? response?.headers : undefined;
 
-          if (!!headers['set-cookie']) {
-            const matchAccess = /folioAccessToken=([^;\s]*)/gi;
-            const matchRefresh = /folioRefreshToken=([^;\s]*)/gi;
-            const cookies = !!headers['set-cookie'] ? headers['set-cookie'] : undefined;
-            let foundAccess;
-            let foundRefresh;
+          const headers = response?.headers;
 
-            if (!!cookies && Array.isArray(cookies)) {
-              let matched;
+          if (headers?.['set-cookie'] && Array.isArray(headers['set-cookie'])) {
+            const extractCookieValue = (token: string) => {
+              const cookie = headers['set-cookie'].find((c: string) => c.startsWith(token));
 
-              for (let i = 0; i < cookies.length; i++) {
-                matched = cookies[i].match(matchAccess);
+              if (cookie) {
+                const valueStartIndex = cookie.indexOf('=') + 1;
+                const valueEndIndex = cookie.indexOf(';', valueStartIndex);
 
-                if (!!matched) {
-                  foundAccess = matched;
-                } else {
-                  matched = cookies[i].match(matchRefresh);
-
-                  if (!!matched) {
-                    foundRefresh = matched;
-                  }
+                if (valueStartIndex > 0 && valueEndIndex > valueStartIndex) {
+                  return cookie.substring(valueStartIndex, valueEndIndex);
                 }
               }
-            } else {
-              foundAccess = cookies.match(matchAccess);
-              foundRefresh = cookies.match(matchRefresh);
-            }
 
-            accessToken = Array.isArray(foundAccess) ? foundAccess[0] : undefined;
-            refreshToken = Array.isArray(foundRefresh) ? foundRefresh[0] : undefined;
+              throw new Error('Invalid cookie');
+            };
+
+            accessToken = extractCookieValue('folioAccessToken');
+            refreshToken = extractCookieValue('folioRefreshToken');
+
           } else {
-            if (!!response?.body?.okapiToken) {
+            if (response?.body?.okapiToken) {
               accessToken = response.body.okapiToken;
-            } else if (!!headers['x-okapi-token']) {
+            } else if (headers?.['x-okapi-token']) {
```

Comment:
```suggestion
            } else if (!!headers?.['x-okapi-token']) {
```
This is Javascript, we need to use the double negate to enforce boolean.